src/main/scala/is/hail/variant/RegionValueVariant.scala

```diff
@@ -0,0 +1,64 @@
+package is.hail.variant
+
+import is.hail.utils._
+
+import is.hail.annotations._
+import is.hail.expr._
+
+class RegionValueVariant(tv: TVariant) extends Variant with View {
+  private val t: TStruct = tv.representation.asInstanceOf[TStruct]
+  private var region: MemoryBuffer = _
+  private var offset: Long = _
+  private val contigIdx: Int = t.fieldIdx("contig")
+  private val startIdx: Int = t.fieldIdx("start")
+  private val refIdx: Int = t.fieldIdx("ref")
+  private val altAllelesIdx: Int = t.fieldIdx("altAlleles")
+  private val tAltAlleles: TArray = t.fieldType(altAllelesIdx).asInstanceOf[TArray]
+  private val tAltAllele: TAltAllele = tAltAlleles.elementType.asInstanceOf[TAltAllele]
+  private val altAllelesView = new ArrayView(tAltAlleles, new RegionValueAltAllele(tAltAllele))
+
+  assert(t.fieldType(contigIdx).required)
+  assert(t.fieldType(startIdx).required)
+  assert(t.fieldType(refIdx).required)
+  assert(tAltAlleles.required)
+  assert(tAltAlleles.elementType.required)
+
+  def setRegion(region: MemoryBuffer, offset: Long) {
+    this.region = region
+    this.offset = offset
+    cachedContig = null
+    cachedRef = null
+    altAllelesView.setRegion(region, t.loadField(region, offset, altAllelesIdx))
+    val a = new Array[AltAllele](altAllelesView.length)
```

Comment:
Would it be useful to move this into the altAlleles() method so that it's not called unless altAlleles is needed? I don't know if there's ever a case where you'd want that, but maybe in the future you'd want to break out nAltAlleles/isBiallelic/altAllele to acoid ever allocating an array for biallelics?

That way you could maybe (in the future?) override nAltAlleles and eventually be able to allocate